crates/cashu/src/amount.rs

```diff
@@ -199,23 +264,31 @@ impl Amount {
     }
 
     /// Checked addition for Amount. Returns None if overflow occurs.
-    pub fn checked_add(self, other: Amount) -> Option<Amount> {
-        self.0.checked_add(other.0).map(Amount)
+    pub fn checked_add(self, other: Amount<()>) -> Option<Amount<()>> {
```

Comment:
@thesimplekid  Pardon my ignorance, but wouldn't it be better if the other amount had the same generic, and we refuse to do the math if the currency units are different? Or the implementor can implement a way to unify the units if possible (from Sat to MSat, for instance).

This is a great chance in my opinion, to reduce confusions when doing math with different units

Reply:
If the other Amount had a unit there would be no need for a generic or a second unit type. This allows us to ensure our units match in the mint where it matters without losing copy and having to introduce clone in even more places throughout the wallet making this a much harder refactor.